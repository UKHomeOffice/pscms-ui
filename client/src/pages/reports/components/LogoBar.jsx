import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import './LogoBar.scss';

export const fullscreen = (report) => report.fullscreen();
export const reload = (report) => report.reload();

const LogoBar = ({ report, visitedPages }) => {
  const { t } = useTranslation();
  return (
    <div className="logo-bar">
      <button
        id="reload"
        type="button"
        onClick={() => {
          // eslint-disable-next-line no-param-reassign
          visitedPages.current = [];
          reload(report);
        }}
      >
        {t('pages.report.logo-bar.reload')}
      </button>
      <button
        id="fullscreen"
        type="button"
        onClick={() => {
          fullscreen(report);
        }}
      >
        {t('pages.report.logo-bar.fullscreen')}
      </button>
    </div>
  );
};

LogoBar.propTypes = {
  report: PropTypes.shape({
    fullscreen: PropTypes.func.isRequired,
  }).isRequired,
  visitedPages: PropTypes.shape({
    current: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
};

export default LogoBar;
