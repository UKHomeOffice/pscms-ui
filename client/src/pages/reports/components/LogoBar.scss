@import '~govuk-frontend/govuk/settings/all';

.logo-bar {
  align-items: center;
  bottom: 0px;
  display: flex;
  font-family: $govuk-font-family-nta;
  font-size: 12px;
  height: 35px;
  position: absolute;
  right: 10px;

  button {
    border: none;
    background-color: transparent;
    color: govuk-colour('dark-grey');
    cursor: pointer;
    font-size: smaller;
    font-weight: $govuk-font-weight-bold;
    padding: 8px;
    text-transform: uppercase;
    &:focus {
      outline: none;
    }
  }

  button {
    &:not(:last-child) {
      padding-right: 0px;
      &:after {
        content: '|';
        margin-left: 8px;
      }
    }
  }
}

@media only screen and (max-width: 960px) {
  .logo-bar {
    display: none;
  }
}
