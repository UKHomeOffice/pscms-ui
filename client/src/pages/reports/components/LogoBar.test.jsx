import React from 'react';
import { shallow } from 'enzyme';
import LogoBar from './LogoBar';

describe('LogoBar', () => {
  const props = {
    report: {
      fullscreen: jest.fn(),
      reload: jest.fn(),
    },
    visitedPages: {
      current: [],
    },
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    expect(wrapper.exists()).toBe(true);
  });

  it('fires reload as expected', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    wrapper.find('#reload').simulate('click');
    expect(props.report.reload).toHaveBeenCalledTimes(1);
  });

  it('resets visitedPages as expected', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    wrapper.find('#reload').simulate('click');
    expect(props.visitedPages.current).toEqual([]);
  });

  it('fires fullscreen as expected', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    wrapper.find('#fullscreen').simulate('click');
    expect(props.report.fullscreen).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<LogoBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
