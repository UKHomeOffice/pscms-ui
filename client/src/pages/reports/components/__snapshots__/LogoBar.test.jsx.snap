// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoBar matches snapshot 1`] = `
<div
  className="logo-bar"
>
  <button
    id="reload"
    onClick={[Function]}
    type="button"
  >
    pages.report.logo-bar.reload
  </button>
  <button
    id="fullscreen"
    onClick={[Function]}
    type="button"
  >
    pages.report.logo-bar.fullscreen
  </button>
</div>
`;
